### PR TITLE
STRATODRVE-5506 fixed pin code circles in enter pin code screen are w…

### PR DESCRIPTION
**Pull request by lhmgrassi**
STRATODRVE-5506 fixed pin code circles in enter pin code screen are white

### CODE REVIEW

File: PasscodeLock/PasscodeLockViewController.swift [hunk 1]
```diff
@@ -93,6 +93,14 @@ open class PasscodeLockViewController: UIViewController, PasscodeLockTypeDelegat
         self.updatePasscodeView()
         self.setupEvents()
     }
+	
+	open override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+		super.traitCollectionDidChange(previousTraitCollection)
+
+		for (index, element) in self.placeholders.enumerated() {
+			self.animatePlacehodlerAtIndex(index, toState: element.currentState)
```
Comment (Tomusm) [inline PasscodeLock/PasscodeLockViewController.swift:101]: Ok, but looks hacky. Is there another alternative ?

Comment (lhmgrassi) [reply PasscodeLock/PasscodeLockViewController.swift:101]: I can try more. But I am afraid it's an iOS bug. The system should care it by itself once we provide both colors (light and dark). The bug is very inconsistent, sometimes it happens sometimes not

Comment (Tomusm) [reply PasscodeLock/PasscodeLockViewController.swift:101]: If I'm not mistaken, the color switch is not automatic for `CALayer`s, which is what is used to determine the border color

Comment (lhmgrassi) [reply PasscodeLock/PasscodeLockViewController.swift:101]: Yes, `CALayer` doesn't support dynamic colors, but even trying `myColor.resolvedColor(with: self.traitCollection).cgColor` didn't solve the problem. The problem is not only in the border sometimes the background is wrong too, as in the picture below
IMG_EA669B6F7F0B-1
: 

The first 2 circles should be blue too. But when I changed from dark to light mode, they just got wrong.

Comment (lhmgrassi) [reply PasscodeLock/PasscodeLockViewController.swift:101]: Yes, `CALayer` doesn't support dynamic colors. Even trying `myColor.resolvedColor(with: self.traitCollection).cgColor` didn't solve the problem. BTW, the problem is not only with the border, sometimes the background color gets wrong too, like in the pic below, where the first 2 circles should also be blue, but something got wrong when changing from dark to light mode. :(

IMG_EA669B6F7F0B-1

**Review by Tomusm — APPROVED** [review]

**Review by mtet88 — APPROVED** [review]

**Pull request merged**